container/container_unix.go

```diff
@@ -283,6 +283,11 @@ func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfi
 		cResources.CpusetMems = resources.CpusetMems
 	}
 	if resources.Memory != 0 {
+		// if memory limit smaller than already set memoryswap limit and doesn't
+		// update the memoryswap limit, then error out.
+		if resources.Memory > cResources.MemorySwap && resources.MemorySwap == 0 {
```

Comment:
@thaJeztah  We already has `resources.Memory > resources.MemorySwap` validation before we get here in https://github.com/docker/docker/blob/master/daemon/daemon_unix.go#L273. So I think it's ok just check if the memoryswap is set here.